community/cypher/cypher/src/test/scala/org/neo4j/cypher/internal/compiler/v3_4/SemanticIndexAcceptanceTest.scala

```diff
@@ -19,73 +19,153 @@
+    )
+    bound <- List("<", "<=", "=", ">", ">=")
+  } {
+    testOperator(bound, valueGen)
+  }
 
   override protected def initTest(): Unit = {
     super.initTest()
     for(_ <- 1 to 1000) createLabeledNode("Label")
   }
 
-  private def changeLastChar(f: Char => Char)(in: String) =
-    if (in.isEmpty) ""
-    else
-      in.substring(0, in.length - 1) + (in.last - 1).toChar
+  /**
+    * Value distribution to test. Allow value generation. Can also provide a slightly smaller
+    * or larger version of a value, which allows testing around a property bound.
+    */
+  case class ValueSetup[T <: Value](name: String, generator:Gen[T], lessThan: T => T, moreThan: T => T)
 
-  private def testOperator(operator: String): Unit = {
+  // GENERATORS
 
-    val queryNotUsingIndex = s"match (n:Label) where n.nonIndexed $operator {prop} return n order by id(n)"
-    val queryUsingIndex = s"match (n:Label) where n.indexed $operator {prop} return n order by id(n)"
+  def longGen: Gen[LongValue] =
+    for (x <- Gen.chooseNum(Long.MinValue+1, Long.MaxValue-1)) yield Values.longValue(x)
 
-    def testValue(queryNotUsingIndex: String, queryUsingIndex: String, value: Any): Unit = {
-      val indexedResult = execute(queryUsingIndex, "prop" -> value)
-      execute(queryNotUsingIndex, "prop" -> value).toList should equal(indexedResult.toList)
-      indexedResult.executionPlanDescription().toString should include("NodeIndexSeek")
-    }
+  def doubleGen: Gen[DoubleValue] =
+    for (x <- arbitrary[Double]) yield Values.doubleValue(x)
 
-    def tester[T](propertyValue: T, prev: T => T, next: T => T): Unit = {
-      graph.inTx {
-        createLabeledNode(Map("nonIndexed" -> propertyValue, "indexed" -> propertyValue), "Label")
+  def textGen: Gen[TextValue] =
+    for (x <- Gen.alphaStr) yield Values.stringValue(x)
 
-        withClue("with TxState") {
-          testValue(queryNotUsingIndex, queryUsingIndex, propertyValue)
-          testValue(queryNotUsingIndex, queryUsingIndex, prev(propertyValue))
-          testValue(queryNotUsingIndex, queryUsingIndex, next(propertyValue))
-        }
-      }
-      withClue("without TxState") {
-        testValue(queryNotUsingIndex, queryUsingIndex, propertyValue)
-        testValue(queryNotUsingIndex, queryUsingIndex, prev(propertyValue))
-        testValue(queryNotUsingIndex, queryUsingIndex, next(propertyValue))
-      }
-    }
+  def pointGen: Gen[PointValue] =
+    for {
+      dimension <- Gen.oneOf(allCRSDimensions)
+      coordinates <- Gen.listOfN(dimension, arbitrary[Double])
+      crs <- Gen.oneOf(allCRS(dimension))
+    } yield Values.pointValue(crs, coordinates:_*)
 
-    test(s"testing long with $operator") {
-      graph.createIndex("Label", "indexed")
-      forAll { propertyValue: Long =>
-        tester(propertyValue, (l: Long) => l - 1L, (l: Long) => l + 1)
-      }
-    }
+  def timeGen: Gen[TimeValue] =
+    for {
+      nanosOfDay <- Gen.chooseNum(1000000000, 86399999999999L - 1000000000)
```

Comment:
Where does this range come from? Why are you not checking values closer to 0? Maybe add a comment about it.

Reply:
Yes, let's improve readability here and add a comment or named constants.